lib/ansible/modules/packaging/os/flatpak.py

```diff
@@ -0,0 +1,264 @@
+    sample: https://sdk.gnome.org/gnome-apps.flatpakrepo
+'''
+
+
+import subprocess
+from ansible.module_utils.basic import AnsibleModule
+
+
+def install_flat(binary, flat):
+    command = "{} install -y --from {}".format(binary, flat)
+    output = flatpak_command(command)
+    if 'error' in output and 'already installed' not in output:
+        return 1, output
+
+    return 0, output
+
+
+def uninstall_flat(binary, flat):
+    # This is a difficult function because it seems there
+    # is no naming convention for the flatpakref to what
+    # the installed flatpak will be named.
+    common_name = parse_flat(flat)
+    command = "{} list --app".format(binary)
+    process = subprocess.Popen(
+        command.split(), stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    output = subprocess.check_output(
+        ("grep", "-i", common_name), stdin=process.stdout)
+    process.wait()
+    installed_flat_name = output.split('/')[0]
+    command = "{} uninstall {}".format(binary, installed_flat_name)
+    output = flatpak_command(command)
+    if 'error' in output and 'not installed' not in output:
+        return 1, output
+
+    return 0, output
+
+
+def parse_remote(remote):
+    name = remote.split('/')[-1]
+    if '.' not in name:
+        return name
+
+    return name.split('.')[0]
+
+
+def parse_flat(name):
+    if 'http://' in name or 'https://' in name:
+        app = name.split('/')[-1].split('.')[0]
+        common_name = app
+    else:
+        common_name = name
+
+    return common_name
+
+
+def add_remote(binary, remote):
+    remote_name = parse_remote(remote)
+    command = "{} remote-add --if-not-exists {} {}".format(
+        binary, remote_name, remote)
+    output = flatpak_command(command)
+    if 'error' in output:
+        return 1, output
+
+    return 0, output
+
+
+def remove_remote(binary, remote):
+    remote_name = parse_remote(remote)
+    command = "{} remote-delete --force {} ".format(binary, remote_name)
+    output = flatpak_command(command)
+    if 'error' in output and 'not found' not in output:
+        return 1, output
+
+    return 0, output
+
+
+def is_present_remote(binary, remote):
+    remote_name = parse_remote(remote).lower() + " "
+    command = "{} remote-list".format(binary)
+    output = flatpak_command(command)
+    if remote_name in output.lower():
+        return True
+
+    return False
+
+
+def is_present_flat(binary, name):
+    command = "{} list --app".format(binary)
+    flat = parse_flat(name).lower()
+    output = flatpak_command(command)
+    if flat in output.lower():
+        return True
+
+    return False
+
+
+def flatpak_command(command):
+    process = subprocess.Popen(
+        command.split(), stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    output = process.communicate()[0]
+
+    return output
+
+
+def main():
+    # This module supports check mode
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=False, default=''),
```

Comment:
This whole block should be changed into:
```python
    module = AnsibleModule(
        argument_spec=dict(
            name=dict(type='str', required=True),
            remote=dict(type='str'),
            state=dict(type='str', default="present", choices=['absent', 'present'])
        ),
        supports_check_mode=True,
        required_if = (
            ('state', 'present', ('repo')),
        ),
    )
```
Assuming that **name** is always required, and **repo** is required when **state == 'present'**